packages/grpc-js/test/test-resolver.ts

```diff
@@ -207,6 +207,64 @@ describe('Name Resolver', () => {
       const resolver = resolverManager.createResolver(target, listener, {});
       resolver.updateResolution();
     });
+    // Created DNS TXT record using TXT sample from https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md
+    // "grpc_config=[{\"serviceConfig\":{\"loadBalancingPolicy\":\"round_robin\",\"methodConfig\":[{\"name\":[{\"service\":\"MyService\",\"method\":\"Foo\"}],\"waitForReady\":true}]}}]"
+    it.skip('Should resolve a name with TXT service config', done => {
+      const target = resolverManager.mapUriDefaultScheme(parseUri('grpctest.kleinsch.com')!)!;
+      const listener: resolverManager.ResolverListener = {
+        onSuccessfulResolution: (
+          addressList: SubchannelAddress[],
+          serviceConfig: ServiceConfig | null,
+          serviceConfigError: StatusObject | null
+        ) => {
+          if (serviceConfig !== null) {
+            assert(
+              serviceConfig.loadBalancingPolicy === 'round_robin',
+              'Should have found round robin LB policy'
+            );
+            done();
+          }
+        },
+        onError: (error: StatusObject) => {
+          done(new Error(`Failed with status ${error.details}`));
+        },
+      };
+      const resolver = resolverManager.createResolver(target, listener, {});
+      resolver.updateResolution();
+    });
+    it.skip(
+      'Should not resolve TXT service config if we disabled service config',
+      (done) => {
+        const target = resolverManager.mapUriDefaultScheme(
+          parseUri('grpctest.kleinsch.com')!
+        )!;
+        let count = 0;
+        const listener: resolverManager.ResolverListener = {
+          onSuccessfulResolution: (
+            addressList: SubchannelAddress[],
+            serviceConfig: ServiceConfig | null,
+            serviceConfigError: StatusObject | null
+          ) => {
+            assert(
+              serviceConfig === null,
+              'Should not have found service config'
+            );
+            count++;
+          },
+          onError: (error: StatusObject) => {
+            done(new Error(`Failed with status ${error.details}`));
+          },
+        };
+        const resolver = resolverManager.createResolver(target, listener, {
+          'grpc.service_config_disable_resolution': 1,
+        });
+        resolver.updateResolution();
+        setTimeout(() => {
+          assert(count === 1, 'Should have only resolved once');
+          done();
+        }, 3_000);
+      }
+    ).timeout(10_000);
```

Comment:
It looks like this is causing an error now that the test is skipped. I think you can avoid that by using a non-arrow function for the test behavior and calling `this.timeout(10_000)`.